ext/sqlite3/database.c

```diff
@@ -717,6 +717,55 @@ static VALUE transaction_active_p(VALUE self)
   return sqlite3_get_autocommit(ctx->db) ? Qfalse : Qtrue;
 }
 
+static int callback_function(void *callback_ary, int count, char **data, char **columns)
+{
+  VALUE * result = (VALUE *)callback_ary;
+  VALUE row_ary = rb_ary_new();
+  int i;
+
+  for (i = 0; i < count; i++) {
+    if (data[i] == NULL){
+      rb_ary_push(row_ary, Qnil);
+    } else {
+      VALUE str = rb_str_new_cstr(data[i]);
+      rb_ary_push(row_ary, str);
+    }
+  }
+
+  rb_ary_push(*result, row_ary);
+
+  return 0;
+}
+
+/* Is invoked by calling db.execute_batch2(sql)
+ *
+ * Executes all statments in a given string separated by semicolons.
+ * If a query is made, all rows will be returned as arrays inside of an array.
+ * If no query is made, an empty array will be returned.
+ */
+static VALUE exec_batch(VALUE self, VALUE sql)
+{
+  sqlite3RubyPtr ctx;
+  int status;
+  VALUE callback_ary = rb_ary_new();
+  char *errMsg;
+  VALUE errexp;
+
+  Data_Get_Struct(self, sqlite3Ruby, ctx);
+  REQUIRE_OPEN_DB(ctx);
+
+  status = sqlite3_exec(ctx->db, StringValuePtr(sql), callback_function, &callback_ary, &errMsg);
```

Comment:
`VALUE` is already a pointer, so you can just do `callback_ary` here and remove the casts in `callback_function`